### ISSUE TITLE
UserRegistry subsystem uses user id and username as if they are the same value

### ISSUE THREAD

**Issue by treo** [OpenLiberty/open-liberty#16165]
This issue originally started out as something that looked like a simple bug, but as I descended deeper and deeper into the code to figure out if there is a good workaround, I found only more and more variants of that same bug.

I ran into the problem while trying to implement a custom `UserRegistry` to be used with the OpenID Provider feature. My first test without enabling that particular feature worked fine. 

When I do enable the feature, it does pull in the ldap registry and federated registry features and now the WSCredentialProvider gets involved after login. So the problem happens even in the login that doesn't touch any of the OpenID / OAuth features.

The core of the problem is simple: Throughout the entire UserRegistry subsystem there are cases where user id and username are used as if they are always the same value.

I guess it hasn't really been noticed yet, because it works with the [BasicRegistry](https://github.com/OpenLiberty/open-liberty/blob/integration/dev/com.ibm.ws.security.registry.basic/src/com/ibm/ws/security/registry/basic/internal/BasicRegistry.java#L578-L589) and the [QuickStartSecurityRegistry](https://github.com/OpenLiberty/open-liberty/blob/integration/dev/com.ibm.ws.security.quickstart/src/com/ibm/ws/security/quickstart/internal/QuickStartSecurityRegistry.java#L162-L173). And it works there because the username and id are actually the same in those implementations.

In the following I show a few cases of that:

#### UsernameAndPasswordLoginModule
According to `UserRegistry.checkPassword` javadoc the method returns a username. https://github.com/OpenLiberty/open-liberty/blob/3a1e114e4c8ff8e686c3e421002eadc80a9236c7/dev/com.ibm.websphere.security/src/com/ibm/websphere/security/UserRegistry.java#L89-L92

But in the `UsernameAndPasswordLoginModule`, it is used as a user id, and then it goes on and tries to get the username (which it already has!) from this id, but because the id is wrong it will fail in an implementation where the id and the username are actually different.
https://github.com/OpenLiberty/open-liberty/blob/3a1e114e4c8ff8e686c3e421002eadc80a9236c7/dev/com.ibm.ws.security.authentication.builtin/src/com/ibm/ws/security/authentication/jaas/modules/UsernameAndPasswordLoginModule.java#L75-L77

#### WSCredentialProvider
It sets `uniqueName` to the user id:
https://github.com/OpenLiberty/open-liberty/blob/integration/dev/com.ibm.ws.security.credentials.wscred/src/com/ibm/ws/security/credentials/wscred/internal/WSCredentialProvider.java#L154-L161

Because of even more variable name confusion, that gets "corrected" because `WIMUserRegistry` actually expects an id instead of a username here. 

#### MembershipBridge
It tries to get `uniqueGroupIds`, which also expects a user id, but then it goes on and tries to get the entity by identifier:
https://github.com/OpenLiberty/open-liberty/blob/integration/dev/com.ibm.ws.security.wim.registry/src/com/ibm/ws/security/wim/registry/util/MembershipBridge.java#L124-L125
However, this only works with usernames!

#### URBridge
Even if I tinker with it i the debugger to force it to use a `uniqueId` identifier instead of a `uniqueName` identifier in that place, it will later fail due to more user id / username confusion in `URBridge.isValidUserOrGroup`:
https://github.com/OpenLiberty/open-liberty/blob/integration/dev/com.ibm.ws.security.wim.core/src/com/ibm/ws/security/wim/adapter/urbridge/URBridge.java#L1094-L1107

As it passes a user id into `UserRegistry.isValidUser` and `UserRegistry.isValidGroup`, both of which expect **names** according to the javadoc.

#### In conclusion 
Because the simple default implementations of `UserRegistry` use the same value for both usernames and user ids, the entire subsystem is riddled with subtle bugs where those values are used interchangeably. 

Until those bugs are fixed, the only way to have a working `UserRegistry` along with more complex features enabled (like `openidConnectServer-1.0`, `federatedRegistry-1.0`), is to use the username as the user id.
